### ISSUE TITLE
Invalid python interpreter path: `/usr/local/bin/python`

### ISSUE THREAD

**Issue by weikanglim** [devcontainers/features#258]
When using `ghcr.io/devcontainers/features/python:1`, the devcontainer opens up with an invalid python interpreter path `/usr/local/bin/python`.

When attempting to debug you get this error:
![image](https://user-images.githubusercontent.com/2322434/199635606-7f94d65e-6bfe-4d9b-baba-4b4fd203e364.png)

After clicking "Select Python interpreter", it shows the invalid path and a detected path:
![image](https://user-images.githubusercontent.com/2322434/199635323-debb60f7-f31b-47c3-8bae-0de677699024.png)

Ctrl + P -> Open Remote Settings (JSON) shows the following settings under `~/.vscode-server/data/Machine`:
```json
{
	"python.defaultInterpreterPath": "/usr/local/bin/python",
	"python.formatting.autopep8Path": "/usr/local/py-utils/bin/autopep8",
	"python.formatting.blackPath": "/usr/local/py-utils/bin/black",
	"python.formatting.yapfPath": "/usr/local/py-utils/bin/yapf",
	"python.linting.banditPath": "/usr/local/py-utils/bin/bandit",
	"python.linting.flake8Path": "/usr/local/py-utils/bin/flake8",
	"python.linting.mypyPath": "/usr/local/py-utils/bin/mypy",
	"python.linting.pycodestylePath": "/usr/local/py-utils/bin/pycodestyle",
	"python.linting.pydocstylePath": "/usr/local/py-utils/bin/pydocstyle",
	"python.linting.pylintPath": "/usr/local/py-utils/bin/pylint"
}
```

I was able to create a minimal repro in https://github.com/weikanglim/devcontainer-python. I have full `devcontainer` build logs if needed.

Originally from https://github.com/Azure/azure-dev/issues/922

**Comment by jkeech** [comment]
@joshspicer it looks like this is coming from https://github.com/devcontainers/features/blob/main/src/python/devcontainer-feature.json#L62-L73. The issue is that based on the options the user specifies (such as [`"version": "os-provided"`](https://github.com/weikanglim/devcontainer-python/blob/main/.devcontainer/devcontainer.json#L6) or tweaking `installPath`), the Python install path can be different. But the devcontainer.json contributions, including the `customizations.vscode` block, are static.

I think we need something for features to be able to dynamically alter the devcontainer.json metadata that they contribute. Or at least for this feature, we need to make sure it always installs to the same path no matter which options are selected.

**Comment by weikanglim** [comment]
@jkeech FWIW the docs seem to suggest that `"version": "os-provided"` is the default setting when unspecified. I tried removing ["version": "os-provided"](https://github.com/weikanglim/devcontainer-python/blob/main/.devcontainer/devcontainer.json#L6) explicitly, and saw the same behavior.

**Comment by Chuxel** [comment]
> @joshspicer it looks like this is coming from https://github.com/devcontainers/features/blob/main/src/python/devcontainer-feature.json#L62-L73. The issue is that based on the options the user specifies (such as [`"version": "os-provided"`](https://github.com/weikanglim/devcontainer-python/blob/main/.devcontainer/devcontainer.json#L6) or tweaking `installPath`), the Python install path can be different. But the devcontainer.json contributions, including the `customizations.vscode` block, are static.
> 
> I think we need something for features to be able to dynamically alter the devcontainer.json metadata that they contribute. Or at least for this feature, we need to make sure it always installs to the same path no matter which options are selected.

When `os-provided` specified the feature could create a symlink from `/usr/local/bin/python` to `/` to resolve this.

**Comment by Chuxel** [comment]
In fact - the current path is just plain wrong in many cases.  The default path is `/usr/local/python`. The path is then updated to soemthing else that looks like got broken somewhere along the way.  This is a bug.

**Comment by Chuxel** [comment]
Self-assigning... I see what is wrong and know how to fix it.  I also see a test bug I'll fix at the same time.

**Comment by Chuxel** [comment]
This should be fixed in the latest released feature (1.0.11)
